ambry-cloud/src/test/java/com/github/ambry/cloud/azure/AzureIntegrationTest.java

```diff
@@ -502,8 +502,12 @@ public void testRepairAfterIncompleteCompaction() throws Exception {
     azureDest.getAzureBlobDataAccessor().purgeBlobs(Collections.singletonList(cloudBlobMetadata));
 
     // Try to delete again (to trigger recovery), verify removed from Cosmos
-    assertFalse("Expected delete to return false",
-        azureDest.deleteBlob(blobId, deletionTime, (short) 0, dummyCloudUpdateValidator));
+    try {
+      assertFalse("Expected delete to return false",
+          azureDest.deleteBlob(blobId, deletionTime, (short) 0, dummyCloudUpdateValidator));
+    } catch (CloudStorageException cex) {
+      assertEquals("Unexpected error code", HttpConstants.StatusCodes.NOTFOUND, cex.getStatusCode());
```

Comment:
This doesn't make sense, you can't expect the method to return false and throw an exception.

Reply:
fixed.